### ISSUE TITLE
ansible-remote provisioner error when playbook looks up environment variable

### ISSUE THREAD

**Issue by Ludachrispeed** [mitchellh/packer#3367]
I am using the ansible-remote provisioner to create an Amazon Machine Image. I've noticed there is a problem if the ansible playbook looks up an environment variable.

Everything works fine if I use the `"ssh_host_key_file"` option of the ansible provisioner. However, if I don't use the `"ssh_host_key_file"` option (and the playbook attempts to look up an environment variable), then there's always a connection issue with the ssh proxy.
- Packer Version
  0.9.0
- Host platform
  OSX 10.11
- Debug log output from `PACKER_LOG=1 packer build template.json`.
  https://gist.github.com/Ludachrispeed/7abcff6480c0998c493c
- The _simplest example template and scripts_ needed to reproduce the bug.
  https://gist.github.com/Ludachrispeed/fab1b5f8166b836ab56d
  See comment at the bottom of the Gist for instructions on running.

**Comment by Ludachrispeed** [comment]
Ok, I've got workaround. If I explicitly disable ansible host key checking in the packer template, my playbook works without the `"ssh_host_key_file"` option. So that's great!

```
"provisioners": [
    {
      "type": "ansible",
      "playbook_file": "./simple.yml",
      "host_alias": "testing",
      "ansible_env_vars": [
        "ANSIBLE_HOST_KEY_CHECKING=False"
      ]
    }
  ]
```

The documentation says that if you don't set the `ansible_env_vars`, it should default to `ANSIBLE_HOST_KEY_CHECKING=False` (https://www.packer.io/docs/provisioners/ansible.html#ansible_env_vars). Perhaps this is the root of the bug?

**Comment by cbednarski** [comment]
Thanks for the report, repro, and workaround!

**Comment by InAnimaTe** [comment]
So I'm not 100% sure but I think I may be having the same problem of sorts... It seems I can't get past connection issues to the proxy meaning ansible just dies everytime. I've tried  solution`#2` to no avail.

You can view the relevant portion of my packer debug log (from when packer connects to the instance to the end) [here](https://tty0.in/lexidiwuci.coffee).

```
   "provisioners": [{                        
       "type": "ansible",                    
       "playbook_file": "./ansible/play.yml",
       "host_alias": "testing",              
       "ansible_env_vars": [                 
         "ANSIBLE_HOST_KEY_CHECKING=False"   
        ]                                    
   }]                                        

```

Any insight on this would be appreciated.

**Comment by InAnimaTe** [comment]
Ohhh wait. I think it might be that 127 happening because 

```
[root@aws-node ~]# which /usr/lib/sftp-server
which: no sftp-server in (/usr/lib)
```

**Comment by bhcleek** [comment]
@Ludachrispeed is this still an issue for you?  It looks like it was fixed in `v0.10.0`.
